### PR TITLE
ShouldBeginWith & ShouldEndWith

### PR DESCRIPTION
**Pull request by bmpj13**
Here's my solution to implement the predicates proposed on #281 and #282. It uses Linq to access the first/last element of the enumerable.

### CODE REVIEW

**Comment by bmpj13** [comment]
Hey guys, here's my solution to ShouldBegin/EndWith. I haven't implemented it with tolerance yet, because I wanted to know what do you think of the solution right now.

Btw, I'm using Linq's First and Last, instead of the First/LastOrDefault, because if it goes to default, then it can lead do some mistakes. For example:
```
[Fact]
public void BeginShouldPass()
{
      int[] target = {};
      target.ShouldBeginWith(0);
}
```
Passes because FirstOrDefault defaults to 0. Using only First, it will throw InvalidOperationException	, stating that the enumerable is empty... I wanted to point it out to know what you guys think :smile:

**Comment by JosephWoodward** [comment]
Sorry it's taken me so long to get around to this, it's been a busy few months! Out of interest, what was your reasoning to go with `ShouldBeginWith` vs `ShouldStartWith`?

File: src/Shouldly/ShouldlyExtensionMethods/ShouldBeEnumerableTestExtensions.cs [hunk 1]
```diff
@@ -43,6 +43,38 @@ public static void ShouldNotContain<T>(this IEnumerable<T> actual, T expected, [
...
+        public static void ShouldBeginWith<T>(this IEnumerable<T> actual, T expected, string customMessage)
+        {
+            ShouldBeginWith(actual, expected, () => customMessage);
+        }
+
+        public static void ShouldBeginWith<T>(this IEnumerable<T> actual, T expected, [InstantHandle] Func<string> customMessage)
+        {
+            if (!actual.First().Equals(expected))
+                throw new ShouldAssertException(new ExpectedActualShouldlyMessage(expected, actual, customMessage).ToString());
+        }
```
Comment (hrai) [inline src/Shouldly/ShouldlyExtensionMethods/ShouldBeEnumerableTestExtensions.cs]: .NET API uses `StartsWith` so perhaps we should follow the heuristics and rename the methods to `ShouldStartWith`

**Comment by JosephWoodward** [comment]
I'm going to close this one off as we've not had much interest it as a feature and agree with Jake that `enumerable.FirstOrDefault().ShouldBe(element);` is clearer. Thanks for putting the pull request together.

**Pull request closed without merging**